state/factory/util.go

```diff
@@ -100,7 +100,7 @@ func protocolPreCommit(ctx context.Context, sr protocol.StateManager) error {
 	if reg, ok := protocol.GetRegistry(ctx); ok {
 		for _, p := range reg.All() {
 			post, ok := p.(protocol.PreCommitter)
-			if ok && sr.ProtocolDirty(p.Name()) {
+			if ok {
```

Comment:
better to double-check, AFAIK `PreCommit()` is meant to be called only when the protocol is dirty, i.e., had some pending change. change to calling it always may change behavior of past blocks